newIDE/app/src/Utils/GDevelopServices/Project.spec.js

```
@@ -2,88 +2,120 @@
         )
       ).toBe(
         '6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png'
       );
       expect(
-        extractFilenameFromProjectResourceUrl(
+        extractDecodedFilenameFromProjectResourceUrl(
           'https://project-resources.gdevelop.io/whatever/some/other-folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png?some=parameter'
         )
       ).toBe(
         '6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea-Happy.png?some=parameter'
       );
     });
+    it('just extracts the filename from an URL which is not from GDevelop Cloud, if encoded', () => {
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl(
+          'example.com/test%20file.png'
+        )
+      ).toBe('test file.png');
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl('example.com/test%20file')
+      ).toBe('test file');
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl(
+          'https://project-resources.gdevelop.io/whatever/some/other%20folder/1/2/_/6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea%20Happy.png'
+        )
+      ).toBe(
+        '6ef87bc678921eb4bfa2d04e5dc6a16b75f7b239f3163d0c5efe64d4cc501711-Pea Happy.png'
+      );
+    });
     it('returns the name given if its not a URL', () => {
-      expect(extractFilenameFromProjectResourceUrl('test.png')).toBe(
+      expect(extractDecodedFilenameFromProjectResourceUrl('test.png')).toBe(
         'test.png'
       );
-      expect(extractFilenameFromProjectResourceUrl('test.png?what')).toBe(
-        'test.png?what'
+      expect(
+        extractDecodedFilenameFromProjectResourceUrl('test.png?what')
+      ).toBe('test.png?what');
+      expect(extractDecodedFilenameFromProjectResourceUrl('test%20.png')).toBe(
+        'test%20.png'
```

Why is it not `'test .png'`?

---

This one is a bit confusing. This function is used to extract almost anything, and if it doesn't match an known url, it just returns the input as is.
Do you know if this was done on purpose, or just a case that should throw an error?